internal/ethapi/api.go

```diff
@@ -1222,8 +1237,8 @@ func (e *revertError) ErrorData() interface{} {
 // useful to execute and retrieve values.
 func (s *BlockChainAPI) Call(ctx context.Context, args TransactionArgs, blockNrOrHash *rpc.BlockNumberOrHash, overrides *StateOverride, blockOverrides *BlockOverrides) (hexutil.Bytes, error) {
 	if blockNrOrHash == nil {
-		latest := rpc.BlockNumberOrHashWithNumber(rpc.LatestBlockNumber)
-		blockNrOrHash = &latest
+		pending := rpc.BlockNumberOrHashWithNumber(rpc.PendingBlockNumber)
+		blockNrOrHash = &pending
```

Comment:
For eth_call I think we still want to default to `latest` to maintain improved compatibility with L1 (though if a user specifies "pending" we also want to support that)